node/pkg/telemetry/loki.go

```diff
@@ -64,28 +84,29 @@ func (logger *ExternalLoggerLoki) log(time time.Time, message json.RawMessage, l
 	}
 	entry := api.Entry{
 		Entry: logproto.Entry{
-			Timestamp: time,
+			Timestamp: ts,
 			Line:      string(bytes),
 		},
 
 		Labels: lokiLabels,
 	}
 
 	select {
-	case logger.c.Chan() <- entry:
+	case logger.bufferedChan <- entry:
 		lokiMessagesSent.Inc()
 	default:
 		lokiMessagesDropped.Inc()
 	}
 
-	// A fatal error exits, which can cause us to lose messages. Flush everything.
+	// A fatal error exits, which can cause us to lose messages. Shut down the worker so it will flush the logs.
 	if level == zapcore.FatalLevel {
-		logger.c.StopNow()
+		logger.stopWorkerWithTimeout()
 	}
 }
 
 func (logger *ExternalLoggerLoki) close() error {
-	logger.c.Stop()
+	// Shut down the worker and wait for it to exit. It has a timeout so we won't wait forever.
+	logger.stopWorkerWithTimeout()
```

Comment:
It looks like the `stopWorkerWithTimeout` doesn't return an error, so this function always returns `nil` and never an `error`. I think the function signature here can be changed to not return anything.
It seems to only be called here:
https://github.com/wormhole-foundation/wormhole/blob/a2fbcc682e8ea4506272409b652d4504690167fd/node/pkg/telemetry/telemetry.go#L89-L91
And that function too can be changed to not have a return value. I looked at the incoming calls to that function and it looks like none of them check the error case anyway.

Alternatively, errors could get propagated out of `logWriter` through `stopWorkerWithTimeout` and into this function.

Reply:
Removed the return value.